tests/test_coreg.py

```diff
@@ -126,3 +131,176 @@ def test_only_paths():
 
     coreg.coregister(reference_raster, to_be_aligned_raster)
+
+
+class TestCoregClass:
+    ref, tba, outlines = load_examples()  # Load example reference, to-be-aligned and mask.
+    mask = outlines.create_mask(ref) == 255
+
+    fit_params = dict(
+        reference_dem=ref.data,
+        dem_to_be_aligned=tba.data,
+        mask=~mask,
+        transform=ref.transform
+    )
+    # Create some 3D coordinates with Z coordinates being 0 to try the apply_pts functions.
+    points = np.array([[1, 2, 3, 4], [1, 2, 3, 4], [0, 0, 0, 0]], dtype="float64").T
+
+    def test_bias(self):
+        warnings.simplefilter("error")
+
+        # Create a bias correction instance
+        biascorr = coreg.BiasCorr()
+        # Fit the bias model to the data
+        biascorr.fit(**self.fit_params)
+
+        # Check that a bias was found.
+        assert biascorr._meta.get("bias") is not None
+        assert biascorr._meta["bias"] != 0.0
+
+        # Copy the bias to see if it changes in the test (it shouldn't)
+        bias = copy.copy(biascorr._meta["bias"])
+
+        # Check that the to_matrix function works as it should
+        matrix = biascorr.to_matrix()
+        assert matrix[2, 3] == bias, matrix
+
+        # Cehck that the first z coordinate is now the bias
+        assert biascorr.apply_pts(self.points)[0, 2] == biascorr._meta["bias"]
+
+        # Apply the model to correct the DEM
+        tba_unbiased = biascorr.apply(self.tba.data, None)
+
+        # Create a new bias correction model
+        biascorr2 = coreg.BiasCorr()
+        # Check that this is indeed a new object
+        assert biascorr is not biascorr2
+        # Fit the corrected DEM to see if the bias will be close to or at zero
+        biascorr2.fit(reference_dem=self.ref.data, dem_to_be_aligned=tba_unbiased, mask=~self.mask)
+        # Test the bias
+        assert abs(biascorr2._meta.get("bias")) < 0.01
+
+        # Check that the original model's bias has not changed (that the _meta dicts are two different objects)
+        assert biascorr._meta["bias"] == bias
+
+    def test_nuth_kaab(self):
+        warnings.simplefilter("error")
+
+        nuth_kaab = coreg.NuthKaab(max_iterations=10)
+
+        # Synthesize a shifted and vertically offset DEM
+        pixel_shift = 2
+        bias = 5
+        shifted_dem = self.ref.data.squeeze().copy()
+        shifted_dem[:, pixel_shift:] = shifted_dem[:, :-pixel_shift]
+        shifted_dem[:, :pixel_shift] = np.nan
+        shifted_dem += bias
+
+        nuth_kaab.fit(self.ref.data.squeeze(), shifted_dem, transform=self.ref.transform)
+
+        assert abs(nuth_kaab._meta["offset_east_px"] - pixel_shift) < 0.03
+        assert abs(nuth_kaab._meta["offset_north_px"]) < 0.03
+        assert abs(nuth_kaab._meta["bias"] + bias) < 0.03
+
+        unshifted_dem = nuth_kaab.apply(shifted_dem, transform=self.ref.transform)
+        diff = np.asarray(self.ref.data.squeeze() - unshifted_dem)
+
```

Comment:
Comments are getting scarce down there, looks like my code ;)

Reply:
Haha! Added better docs in b569902